app/models/audio_version_template.rb

```diff
@@ -43,42 +43,58 @@ def touch_audio_versions
   end
 
   def validate_audio_version(audio_version)
-    [audio_file_count_errors(audio_version),
-     audio_length_errors(audio_version)].join(' ').strip
+    audio_file_count_errors(audio_version) || audio_length_errors(audio_version)
+  end
+
+  def validate_audio_file(audio_file)
+    audio_file_content_type_errors(audio_file)
   end
 
   private
 
   def audio_file_count_errors(audio_version)
-    file_count_errors = ''
     num_audio_files = audio_version.audio_files.count
     if !segment_count.nil? && audio_version.audio_files.count != segment_count
-      file_count_errors << "Audio version #{audio_version.label} has #{num_audio_files} " +
-                           "audio files but must have #{segment_count} segments."
+      "Audio version #{audio_version.label} has #{num_audio_files} audio " +
+        "files but must have #{segment_count} segments."
     end
-    file_count_errors
   end
 
   def audio_length_errors(audio_version)
-    length_errors = ''
-    shorter_than_min = audio_version.length < length_minimum
+    shorter_than_min = length_minimum != 0 && audio_version.length < length_minimum
     longer_than_max = length_maximum != 0 && audio_version.length > length_maximum
     if shorter_than_min || longer_than_max
-      length_errors << "Audio version '#{audio_version.label}' is " +
-                       "#{audio_version.length.to_time_string} long but must be "
+      length_errors = "Audio version '#{audio_version.label}' is " +
+                      "#{audio_version.length.to_time_string} long but must be "
       length_errors << if length_maximum != 0
                          "#{length_minimum.to_time_string} - " +
                          "#{length_maximum.to_time_string} long."
                        else
                          "more than #{length_minimum.to_time_string} long."
                        end
+      length_errors
     end
-    length_errors
   end
 
   def max_is_greater_than_min_if_set
     if length_maximum != 0 && length_minimum > length_maximum
       errors.add(:length_mininum, 'must be less than or equal to length maximum')
     end
   end
+
+  def audio_file_content_type_errors(audio_file)
+    if content_type.present?
+      tpl_type = content_type.split('/').first
+      file_type = audio_file.content_type.try(:split, '/').try(:first)
+      if content_type == 'audio/mpeg' && audio_file.content_type != 'audio/mpeg'
```

Comment:
This is a little problematic.  An mp2 has `content_type = 'audio/mpeg'` but `layer = 2`.  But know that the template `content_type` is audio/mpeg isn't actually specific enough to designate layer.  So for now, just letting them pass validation.

Reply:
yeah, at some point we'll add more specific validations at a template or distro level, and that will be less of an issue - this is more about getting non-audio/video to work than making audio validations awesome.

(should that check for them matching as opposed to both checked against 'audio/mpeg' string?)

Reply:
Not sure i follow on your (2nd comment) there.  You're saying this function should just check that the first half of the mime type matches?  (`audio/` or `video/`)

Reply:
I meant something like:
`if content_type == 'audio/mpeg' && content_type != audio_file.content_type`

since eventually we want to implement for all types, and will become something like
`if content_type != audio_file.content_type`

Reply:
very very minor point

Reply:
ah, i gotcha.  One minor complication to that ... the current `audio_file.content_type` reflects the content_type of the original file.  Not necessarily any transcoded versions of the file (which may satisfy the audio_version_template).

But transcoding offers many, many complications anyways!